mlir/include/mlir/Interfaces/ViewLikeInterface.td

```diff
@@ -414,4 +414,44 @@ def OffsetSizeAndStrideOpInterface : OpInterface<"OffsetSizeAndStrideOpInterface
   }];
 }
 
+def DistinctObjectsInterface : OpInterface<"DistinctObjectsInterface"> {
```

Comment:
Can we start off with a Trait instead of an interface? And generalize to a full-blown interface if we need it. (Right now, this is just passing through all operands + results.)